### PR TITLE
feat(medusa, medusa-js, medusa-react): Implement Sales Channel deletion

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Support sales channel deletion in medusa, medusa-js and medusa-react

**How**

By implementing a new endpoint and the associated service method as well as the repository methods.

Medusa-js new deletion method in the resource

Medusa-react new hook in the mutations

**Tests**
- Endpoint test
- Service test
- Integration test
- Hook tests

Fixes CORE-278

### CODE REVIEW

File: packages/medusa/src/services/__tests__/sales-channel.ts [hunk 1]
```diff
@@ -92,4 +92,61 @@ describe("SalesChannelService", () => {
...
+          id:
+            typeof queryOrId === "string"
+              ? queryOrId
+              : ((queryOrId?.where as FindConditions<SalesChannel>)?.id ?? IdMap.getId("sc_adjhlukiaeswhfae")),
+          ...salesChannelData
+        })
+      }),
+      softRemove: jest.fn().mockImplementation((id: string): any => {
+        return Promise.resolve()
+      }),
```
Comment (pKorsholm) [inline packages/medusa/src/services/__tests__/sales-channel.ts:115]: should this be added to the repository in the outer describe and that repo mock be used instead?

**Review by pKorsholm — COMMENTED** [review]
Generally looks good! I have a few comments 😄

File: packages/medusa/src/api/routes/admin/sales-channels/delete-sales-channel.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ *               type: boolean
+ */
+export default async (req: Request, res: Response): Promise<void> => {
+  const { id } = req.params
+
+  const salesChannelService: SalesChannelService = req.scope.resolve("salesChannelService")
+  await salesChannelService.delete(id)
+  res.json({
+    id,
+    object: "salesChannel",
```
[the remark this answers is not in the document]
Comment (adrien2p) [reply packages/medusa/src/api/routes/admin/sales-channels/delete-sales-channel.ts]: I moved to hyphenated here at the moment 👍

File: packages/medusa-react/src/hooks/admin/sales-channels/queries.ts [hunk 3]
```diff
@@ -13,6 +13,11 @@ export const adminSalesChannelsKeys = queryKeysFactory(
 
 type SalesChannelsQueryKeys = typeof adminSalesChannelsKeys
 
+/**
+ * @experimental This feature is still under development and can't be used in production
```
Comment (olivermrbl) [inline packages/medusa-react/src/hooks/admin/sales-channels/queries.ts:17]: todo: Please use exact same format as seen in [this PR](https://github.com/medusajs/medusa/pull/1811/files)

Note, we'll add it to the Code Convention doc 👍

**Review by olivermrbl — COMMENTED** [review]
LGTM! Couple of comments 👍

**Review by fPolic — APPROVED** [review]
LGTM! 👍

**Review by srindom — APPROVED** [review]
LGTM - with comments!

**Pull request merged**